Telink w91

s07641069 opened the pull request:
Temp PR to check CI
DO NOT MERGE

serhiiSalamakha committed:
samples: add sw irq for W91

Added sw irq for W91

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
samples: update basic sample for D25 core (W91)

Updated basic/sample for D25 core (W91)

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/439208c46aa98297066cba042c605770c07280d8 (develop) | https://github.com/telink-semi/hal_telink/commit/ca708d762a252fafeb7ff382e6053bd5caeb8966 (telink_w91) | [telink-semi/hal_telink@439208c4..ca708d76](https://github.com/telink-semi/hal_telink/compare/439208c46aa98297066cba042c605770c07280d8..ca708d762a252fafeb7ff382e6053bd5caeb8966) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

s07641069 committed:
boards: riscv: telink: Add tlsr9118bdk40d board

soc: riscv: telink_b9x: Add w91 SoC

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
boards: riscv: telink: Add w91 blinky sample

soc: riscv: telink_b9x: Adopt start.S file

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
soc: riscv: telink_b9x: Use D25 core for Zephyr

- Move Zephyr from N22 into D25 core
- Add start N22 core (Simple/FreeRTOS) from 0x80300000 flash addr

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
soc: riscv: telink_b9x: Fixed FreeRTOS run on N22 core

- Remove SRAM usage due to conflict with FreeRTOS
- Add print of running core
- Update start.S to latest version from Simple project

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
soc: riscv: telink_b9x: SPI flash in quad mode.

Added SPI flash regularand quad modes.

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
soc: riscv: telink_b9x: Tested runing in paralell with FreeRTOS

Disable conflicted code

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
soc: riscv: telink_b9x: Use k_msleep instead of nops

Set correct SYS_CLOCK_HW_CYCLES_PER_SEC

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
soc: riscv: telink_b9x: Disable Software reset defines

TBD reset D25 core or reset via FreeRTOS WDG

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
soc: riscv: telink_b9x: Fix DLM start addr

Align memory usage with FreeRTOS

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
CI: Add w91 blinky build and fix failed jobs

Fix CI jobs

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
CI: Fix failed jobs

Fix CI jobs

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>
